### ISSUE TITLE
[Bug]: [resourcegroup]different behavior between 2.2.0 and master when trying to create rg with dup name

### ISSUE THREAD

**Issue by yanliang567** [milvus-io/milvus#23912]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.8 Vs master-0506
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

in 2.2.0
error msg: resource group already exist
in 2.3.0
success

### Expected Behavior

same behavior: fail to create rg with dup name

### Steps To Reproduce

```markdown
run testcase: test_create_rg_dup_name
```

**Comment by yanliang567** [comment]
one more different error msg when trying to create max+1 resource groups
milvus 2.2.0:  pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=failed to create resource group, err=resource group num reach limit 1024)>
milvus master-:  pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=resource group num reach limit 1024)>

**Comment by weiliu1031** [comment]
> one more different error msg when trying to create max+1 resource groups(run test test_create_max_number_rgs) milvus 2.2.0: pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=failed to create resource group, err=resource group num reach limit 1024)> milvus master-: pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=resource group num reach limit 1024)>

Since 2.3, we introduced the merr, to refine the err msg,  and the err msg has changed, which is by design.

**Comment by weiliu1031** [comment]
please verify this

**Comment by yanliang567** [comment]
verified fixed on master-20230510-9715a850
